hashes.py

```diff
@@ -0,0 +1,128 @@
+"""
+Artem
+
+The requirements for a Hash Function:
+
+1. A hash function must be consistent(deterministic). Every time it receives the sam input (like "aqua"), it must return the same output (like 4). If it's not deterministic, it's no a hash function. Same input will always return the same output. 
+
+2. Different input data should return different numbers to minimize collisions. 
+
+3. A hash function must return numbers that are within a specific range. 
+
+Hashing: blockchain, saving passwords in the BE, SHA-256 (256 bits limit), SHA-512 (used in class), etc. 
+SHA: Secure Hashing Algo
+
+"""
+# Magic Function: 
+# takes a word --> returns the index fo where that word is located in some array of
+# book --> 1
+# egypt --> 4
+# fake_word --> None
+# this fn is O(1)
+
+# HASH FUNCTIONS
+# Any string input ---> A specific number (within some range)
+# MOST IMPORTANT!
+# This fn must be deterministic
+# Same input will alwasy return the same output
+# Non-reversible
+# Avoid collisions
+
+hash_table = [None] * 8 # 8 slots, all initialized to None
+
+# O(n) for length of key
+def my_hash(self):
+    # take every character in the string, and convert character to number
+    # Convert each character into UTF-8 numbers (it allows for more characters than ASCII & Unicode)
+    string_utf = self.encode()
+    
```

Comment:
Awesome move, converting to utf-8, it's good to be aware of that string format, as well as making sure you can handle a wider range of strings and characters- you never know what someone will want to use as a key.

Reply:
Yeap, specially these days, we do a lot of transactions/businesses with China and other countries : )